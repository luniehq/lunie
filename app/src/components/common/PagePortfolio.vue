<template>
  <TmPage :sign-in-required="true">
    <template slot="signInRequired">
      <TmBalance />
      <DelegationsOverview />
      <Undelegations />
    </template>
  </TmPage>
</template>

<script>
import DelegationsOverview from "staking/DelegationsOverview"
import TmBalance from "common/TmBalance"
import TmPage from "common/TmPage"
import Undelegations from "staking/Undelegations"

export default {
  name: `page-portfolio`,
  components: {
    TmBalance,
    TmPage,
    Undelegations,
    DelegationsOverview,
  },
}
</script>
