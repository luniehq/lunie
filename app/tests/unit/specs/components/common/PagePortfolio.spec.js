import PagePortfolio from "common/PagePortfolio"
import { shallowMount } from "@vue/test-utils"

describe(`PagePortfolio`, () => {
  let wrapper, $store

  const state = {
    session: { experimentalMode: false },
  }

  beforeEach(() => {
    $store = {
      state,
    }

    wrapper = shallowMount(PagePortfolio, {
      mocks: {
        $store,
      },
    })
  })

  it("should display the portfolio page", async () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
