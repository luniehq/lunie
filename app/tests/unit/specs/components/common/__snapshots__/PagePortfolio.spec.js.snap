// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePortfolio should display the portfolio page 1`] = `
<tmpage-stub
  emptysubtitle=""
  emptytitle="No data"
  loaderpath=""
  signinrequired="true"
>
  <template>
    <tmbalance-stub />
     
    <delegationsoverview-stub />
     
    <undelegations-stub />
  </template>
</tmpage-stub>
`;
